src/attr/_make.py

```diff
@@ -2605,10 +2595,8 @@ def from_counting_attr(cls, name, ca, type=None):
     @property
     def cmp(self):
         """
-        Simulate the presence of a cmp attribute and warn.
+        Simulate the presence of a cmp attribute.
         """
-        warnings.warn(_CMP_DEPRECATION, DeprecationWarning, stacklevel=2)
```

Comment:
I've been thinking and I think we should keep this deprecation.

The behavior of cmp is rather muddy, because it's a derivation of eq and order.

Now cmp becomes the officially blessed way of setting eq and order but this attribute doesn't make sense to me.

Especially because it returns a bool, while eq and order can be callables.

Does that make sense to you?

Reply:
I can see where you're coming from.  We have three arguments: `eq`, `order` and `cmp`, so one of them is redundant for sure.  At the same time, `eq` and `order` aren't entirely *clean*, because `order` requires `eq`.

My preference would be to undeprecate `cmp` to avoid `attr.ib(eq='str.lower', order='eq.lower')`. I think `attr.ib(cmp='eq.lower')` feels and looks a bit nicer.

For the record, in my applications I will use `attr.ib(eq=key_func(... ), order=False)`. Very often `order` won't make sense with complex objects. In fact, I'm a lot more annoyed by `order` being `True` by default than I am by `cmp` vs `eq`.

Having said that, I completely understand that `cmp` deprecation has been on the cards for a long time, and that undeprecating it may feel like a regression.  Happy to follow your lead.

Reply:
I think/hope we've misunderstood each other!

What I mean is that we should undeprecate cmp for `attr.s`/`attr.attrs` (and add it to `attr.define` for that matter, but that's a different PR) since it's syntactic sugar for setting eq/order and very useful.

---

But I want to remove it eventually from the Attribute class. That's why I've commented specifically on the change in the cmp property.

Reply:
Oh I see, you want to keep the deprecation specifically in the `cmp` property of `Attribute` - not else where? Sorry, my bad.  Would we keep it in `Attribute`'s `init`?

> Especially because it returns a bool, while eq and order can be callables.

`eq` and `order` are also booleans. The callables are stored under `eq_key` and `order_key`.

Reply:
Do we need a new deprecation text?

```python
_CMP_DEPRECATION = (
    "The usage of `cmp` is deprecated and will be removed on or after "
    "2021-06-01.  Please use `eq` and `order` instead."
)
```

Reply:
If it's not used, probably not?

Reply:
> Would we keep it in Attribute's init?

You mean the deprecation? Yes. I would like to remove cmp from Attribute for good eventually. I want it only to be a syntactic sugar, nothing more.

Reply:
All done (I think). Apologies for the confusion.